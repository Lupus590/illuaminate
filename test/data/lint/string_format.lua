-- config: (lint (only (stdlib:string-format)))

local _ = string.format("%s %.2f % +s") -- ok
local _ = string.format("%?")

local _ = string.format("%2%")
