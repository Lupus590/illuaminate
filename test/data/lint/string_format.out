string_format.lua:[3:11-3:39]: Format string takes 3 parameters, but only given 0 arguments. [stdlib:string-format]
   │
 3 │ local _ = string.format("%s %.2f % +s") -- ok
   │           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
string_format.lua:[4:25-4:28]: Unknown specifier '%?' [stdlib:string-format]
   │
 4 │ local _ = string.format("%?")
   │                         ^^^^
string_format.lua:[6:25-6:29]: Redundant flags to format string '%' [stdlib:string-format]
   │
 6 │ local _ = string.format("%2%")
   │                         ^^^^^
No errors and 3 warnings
